pkg/repository/stock/stockRepositoryImpl.go

```diff
@@ -93,20 +103,50 @@ func (r *stockRepositoryImpl) Delete(ctx context.Context, id string) (*entities.
 		return nil, errors.New(fmt.Sprintf("failed to delete stock"))
 	}
 
-	return stock.ToStockEntity(), nil
+	stockEntity := stock.ToStockEntity()
+	r.SetStockSubAttributes(stockEntity, sp)
+	return stockEntity, nil
 }
 
-func ToStockModel(e *entities.Stock) *model.Stock {
+func (r *stockRepositoryImpl) ToStockModel(e *entities.Stock) *model.Stock {
 	return &model.Stock{
 		ProductID: e.ProductID,
 		Quantity:  e.Quantity,
 	}
 }
 
-func ToStockModelRes(e *entities.Stock) *modelRes.Stock {
+func (r *stockRepositoryImpl) ToStockModelRes(e *entities.Stock) *modelRes.Stock {
 	return &modelRes.Stock{
 		StockID:   e.StockID,
 		ProductID: e.ProductID,
 		Quantity:  e.Quantity,
 	}
 }
+
+func (r *stockRepositoryImpl) SetStockSubAttributes(stockData any, sp trace.Span) {
+	if stocks, ok := stockData.(*[]entities.Stock); ok {
+		var stockIDs []string
```

Comment:
you can "Pre-allocate" slices for good performance, just like this
[https://stackoverflow.com/questions/45423667/what-is-the-point-in-setting-a-slices-capacity](url)

Reply:
ok, Thx!